homeassistant/components/modbus/__init__.py

```diff
@@ -179,9 +178,10 @@
         vol.Optional(CONF_SCALE, default=1): number_validator,
         vol.Optional(CONF_OFFSET, default=0): number_validator,
         vol.Optional(CONF_PRECISION, default=0): cv.positive_int,
-        vol.Optional(CONF_SWAP, default=CONF_SWAP_NONE): vol.In(
+        vol.Optional(
+            CONF_SWAP,
+        ): vol.In(
             [
-                CONF_SWAP_NONE,
```

Comment:
Removing "none" as a valid value is imo a breaking change as a user could specify something like this in yaml:

```yaml
swap: none
```

Sure I think that not many users will specify default values but it could happen. I think we should mark it as breaking change, or what do you think?

Reply:
When would "none" be used, compared to the others?

Reply:
Basically never, we used it internally earlier...

The only scenario is a user that used "swap: xyz" and want to remove it by doing "swap: none", but it is just as easy to remove the whole line.

Reply:
The internal use earlier, was due to some not efficient code, and that have been removed a while ago.

The user will not get a config error (swap: none) is illegal, which reminds him to remove it.

This is an important change for me (as maintainer) because it simplified an upcoming config_flow (along with a number of other changes).